nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/trash/test/AbstractTestTrashService.java

```diff
@@ -257,9 +266,13 @@ public void testUntrashChildren() {
         doc1 = session.getDocument(new IdRef(doc1.getId()));
         doc2 = session.getDocument(new IdRef(doc2.getId()));
         assertFalse(fold.isTrashed());
-        // children done by async BulkLifeCycleChangeListener
+        // children recursively untrashed
         assertFalse(doc1.isTrashed());
         assertFalse(doc2.isTrashed());
+        // fold is unmangled
+        assertFalse(trashService.isMangledName(fold.getName()));
+        // doc1 is recursively unmangled
+        assertFalse(trashService.isMangledName(doc1.getName()));
```

Comment:
I checked it was not mangled (mangling is not recursive). 
It shouldn't have moved when I untrash its parent: /fold.
Or is it worth the check ?

Reply:
Maybe worth checking that the `untrash` action did nothing on the doc2 🤷‍♂

Reply:
since we already check it's untrashed, I added a check on the name to prove it didn't change
```
        // doc2 stays unmangled
        assertFalse(trashService.isMangledName(doc2.getName()));
```